pandas/core/indexes/multi.py

```diff
@@ -694,10 +734,9 @@ def _set_codes(self, codes, level=None, copy=False, validate=True,
                     level_codes, lev, copy=copy)._shallow_copy()
             new_codes = FrozenList(new_codes)
 
-        if verify_integrity:
-            self._verify_integrity(codes=new_codes)
-
+        new_codes = self._verify_integrity(codes=new_codes)
```

Comment:
shouldn't this only be called if verfiy_integrity?

Reply:
Yes, changed